src/internal_events/dnstap.rs

```diff
@@ -26,3 +26,20 @@ impl<'a> InternalEvent for DnstapParseError<'a> {
         counter!("parse_errors_total", 1);
     }
 }
+
+#[derive(Debug)]
+pub(crate) struct DnstapParseWarning<'a> {
+    pub error: &'a str,
```

Comment:
I _believe_ this can be defined as 

```Rust
pub error: E,
```

Where `E : std::fmt::Display` , lots of examples for instance `HostMetricsScrapeDetailError`

That way you don't have to call `to_string().as_str()`

Reply:
Nice.